service/src/test/java/bio/terra/cbas/pact/SamPactTest.java

```diff
@@ -0,0 +1,126 @@
+package bio.terra.cbas.pact;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.PactSpecVersion;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.cbas.config.SamServerConfiguration;
+import bio.terra.cbas.dependencies.sam.SamClient;
+import bio.terra.cbas.dependencies.sam.SamService;
+import bio.terra.common.iam.BearerToken;
+import java.util.Map;
+import org.broadinstitute.dsde.workbench.client.sam.model.UserStatusInfo;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+class SamPactTest {
+  static final String dummyWorkspaceId = "15f36863-30a5-4cab-91f7-52be439f1175";
+  private SamService samService;
+
+  @BeforeEach
+  void setUp() {
+    // Without this setup, the HttpClient throws a "No thread-bound request found" error
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    // Set the mock request as the current request context
+    RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+  }
+
+  private void initSamService(MockServer mockServer) {
+    SamServerConfiguration samConfig =
+        new SamServerConfiguration(mockServer.getUrl(), false, dummyWorkspaceId, true);
+    SamClient samClient = new SamClient(samConfig);
+    samService = new SamService(samClient, new BearerToken("accessToken"));
+  }
+
+  @Pact(consumer = "cbas-consumer", provider = "sam-provider")
```

Comment:
this naming convention - `${SERVICE}-consumer` or `${SERVICE}-provider` - is flawed, despite it being the most common naming convention. I'm working on getting services to rename their contracts. We should all just use the service name: `cbas` in this case.

You may have to keep the provider name `sam-provider` until they change their code, unfortunately...

Reply:
(the reason it is flawed is because there are certain pieces of functionality in the Pact broker that rely on identifying a service regardless of whether it is playing the role of consumer or provider. that functionality is broken for all the services that use the flawed naming convention)

Reply:
I'm saying: there's really no reason for us to contribute to this flawed practice by perpetuating it.

Reply:
I can easily change cbas-consumer to cbas, but I cannot change sam-provider to sam as I will hit check_for_potential_duplicate_pacticipant_names rule. And this change needs to be sync'd with everybody on DSP pact.

Reply:
Yeah, agreed. If we can change `cbas-consumer` to `cbas` that would be great.

Sorry for sticking on this minor point. This has been an uphill battle since we started using Pact.

Reply:
np, I have already tested. Closing this conversation then.